resource/group/group_test.go

```diff
@@ -73,400 +73,597 @@ func init() {
-	g := group.NewGroup(m)
-	g.GID = grp.Gid
-	g.Name = grp.Name
-	g.State = group.StatePresent
-
-	m.On("LookupGroup", g.Name).Return(grp, user.UnknownGroupError(""))
-	m.On("LookupGroupID", g.GID).Return(grp, user.UnknownGroupIdError(""))
-	m.On("AddGroup", g.Name, g.GID).Return(fmt.Errorf(""))
-	status, err := g.Apply()
-
-	m.AssertCalled(t, "AddGroup", g.Name, g.GID)
-	assert.EqualError(t, err, fmt.Sprintf(""))
-	assert.Equal(t, resource.StatusFatal, status.StatusCode())
-	assert.Equal(t, fmt.Sprintf("error adding group %s with gid %s", g.Name, g.GID), status.Messages()[0])
-}
-
-func TestApplyAddGroupNotAdded(t *testing.T) {
-	t.Parallel()
-
-	grp := &user.Group{
-		Name: fakeName,
-		Gid:  fakeGid,
-	}
-	m := &MockSystem{}
-	g := group.NewGroup(m)
-	g.GID = grp.Gid
-	g.Name = grp.Name
-	g.State = group.StatePresent
-
-	m.On("LookupGroup", g.Name).Return(grp, nil)
-	m.On("LookupGroupID", g.GID).Return(grp, nil)
-	m.On("AddGroup", g.Name, g.GID).Return(nil)
-	status, err := g.Apply()
-
-	m.AssertNotCalled(t, "AddGroup", g.Name, g.GID)
-	assert.EqualError(t, err, fmt.Sprintf("will not attempt add: group %s with gid %s", g.Name, g.GID))
-	assert.Equal(t, resource.StatusCantChange, status.StatusCode())
+	})
 }
 
-func TestApplyDeleteGroupErrorDeleting(t *testing.T) {
+// TestSetGroupAddOptions tests the options provided for
+// adding a group are added to the options map correctly
+func TestSetGroupAddOptions(t *testing.T) {
 	t.Parallel()
 
-	grp := &user.Group{
-		Name: fakeName,
-		Gid:  fakeGid,
-	}
-	m := &MockSystem{}
-	g := group.NewGroup(m)
-	g.GID = grp.Gid
-	g.Name = grp.Name
-	g.State = group.StateAbsent
-
-	m.On("LookupGroup", g.Name).Return(grp, nil)
-	m.On("LookupGroupID", g.GID).Return(grp, nil)
-	m.On("DelGroup", g.Name).Return(fmt.Errorf(""))
-	status, err := g.Apply()
-
-	m.AssertCalled(t, "DelGroup", g.Name)
-	assert.EqualError(t, err, fmt.Sprintf(""))
-	assert.Equal(t, resource.StatusFatal, status.StatusCode())
-	assert.Equal(t, fmt.Sprintf("error deleting group %s with gid %s", g.Name, g.GID), status.Messages()[0])
-}
-
-func TestApplyDeleteGroupNotDeleted(t *testing.T) {
-	t.Parallel()
+	t.Run("all supported add options", func(t *testing.T) {
+		g := group.NewGroup(new(group.System))
+		g.Name = fakeName
+		g.GID = fakeGid
+		options := group.SetGroupAddOptions(g)
 
-	grp1 := &user.Group{
-		Name: fakeName,
-		Gid:  fakeGid,
-	}
-	grp2 := &user.Group{
-		Name: currName,
-		Gid:  currGid,
-	}
-	m := &MockSystem{}
-	g := group.NewGroup(m)
-	g.GID = grp2.Gid
-	g.Name = grp1.Name
-	g.State = group.StateAbsent
-
-	m.On("LookupGroup", g.Name).Return(grp1, nil)
-	m.On("LookupGroupID", g.GID).Return(grp2, nil)
-	m.On("DelGroup", g.Name).Return(nil)
-	status, err := g.Apply()
-
-	m.AssertNotCalled(t, "DelGroup", g.Name)
-	assert.EqualError(t, err, fmt.Sprintf("will not attempt delete: group %s with gid %s", g.Name, g.GID))
-	assert.Equal(t, resource.StatusCantChange, status.StatusCode())
-}
+		assert.Equal(t, g.GID, options["gid"])
```

Comment:
why is this a map of strings instead of a real struct? It might make more sense for it to be:

```go
type GroupAddOptions struct {
    GID uint32
}
```